### PR TITLE
Traktmixxx skin

### PR DESCRIPTION
**Pull request by intialonso**
Hi everybody!
Using my quarantine time I forked and edited the amazing theme RoundCorners from @22degrees, I liked a lot the interface of the new Traktor DJ 2, so I wanted to create a similar skin, RoundCorners proved to be the most completed and best explained theme available, so I took it as the base for this project:

- Not so rounded any more.
- Dark color scheme with bright highlight colors.
- More icons, less text.
- New buttons and knobs.

There are still some things to edit (Waveform editor, Waveform samples, Tooltips, Right click menus, Configuration overlays) but I think it is already usable. Let me know if you find any bugs.

Also I want to ask, are the icons in the library tree view hardcoded? I failed to find a way to change them.

preferences_preview_screenshot

### CODE REVIEW

**Comment by uklotzde** [comment]
Welcome and thank you for responding so promptly :)

The icons in the side bar are fixed afaik. @ronso0 You may proof me wrong.

File: res/skins/Traktmixxx/.gitignore [hunk 1]
```diff
@@ -0,0 +1,16 @@
+RoundedCorners.sublime-project
```
Comment (uklotzde) [inline res/skins/Traktmixxx/.gitignore]: Please revert those changes (doing a force push) and instead extend your global .gitignore in your home directory.

**Comment by uklotzde** [comment]
I have noticed some resolution dependent files. @ronso0 May I assume that all our current skins are resolution independent now?

**Comment by uklotzde** [comment]
The pre-commit hook complains about missing newlines at the end of files. Please keep on I on those.

We also need obtain your consent by signing the contributor agreement: https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ

Otherwise we would not be able to distribute any code contributions. Better do this now so we won't forget about it later.

File: res/skins/Traktmixxx/license.txt [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<!--
+"Traktmixxx", Skin for Mixxx
+www.mixxx.org
+Copyright (C) 2015-2020 Inti Alonso (https://github.com/intialonso), built over "Roundcorners", the amazing work of 22degrees (https://github.com/22degrees)
```
Comment (uklotzde) [inline res/skins/Traktmixxx/license.txt]: "RoundCorners" and it should include the full URL to the GitHub repository. Adding an additional link to the forum post https://www.mixxx.org/forums/viewtopic.php?f=8&t=9443 doesn't hurt.

I would suggest to separate your own copyright state from the attribution.

Comment (intialonso) [reply res/skins/Traktmixxx/license.txt]: Just signed the agreement. 
Im reading about the newlines thing, Im working on Windows, how can I batch fix the issue?
Regarding the license do you mean just puting another line? Like this:

> Copyright (C) 2015-2020 by Inti Alonso https://github.com/intialonso
Built over "RoundCorners", by 22degrees https://github.com/22degrees/RoundCorners https://www.mixxx.org/forums/viewtopic.php?f=8&t=9443

Im sorry if everything is a little messed, Im more a designer than a programer, so a lot of details confuse me.

**Comment by ronso0** [comment]
> I have noticed some resolution dependent files. @ronso0 May I assume that all our current skins are resolution independent now?

If you mean: all our skins can expand to fill the screen? yes
All our skins have a minimum size, 1024 x 600 px, LateNight is for 1280+
Except in Shade we use vector graphics that scale nicely.

**Comment by Be-ing** [comment]
Hi, thanks for submitting this. However, I'm not in favor of including this skin in Mixxx. Mixxx is its own software; we're not trying to verbatim clone any other software. We find inspiration in other software where they have good ideas, but we do not try to imitate them.

Additionally, there would need to be a _lot_ of work for this to be included in Mixxx:
  * All graphics must be vectorized, except level meters because rapidly rerendering SVGs so frequently causes performance problems
  * 4 decks don't come close to fitting on my screen. Even with 2 decks there is no vertical space for the library and effects. I use a 3840x2160 screen with a 3x scale factor.
  * Effects, loop, and beatjump controls look like they were designed for Mixxx 2.0
  * Overview waveforms don't support RGB colors

**Comment by uklotzde** [comment]
@Be I have to agree. Mimicking another application is probably not our intention and might cause confusion among users. I didn't consider the fact that this skin is a visual clone of the Traktor DJ 2 look.

**Comment by Holzhaus** [comment]
The similarity with the Traktor DJ Pro skin in not that eye-catching, so I don't think it's a problem:

This Skin
Traktor DJ Pro

However, I agree that we shouldn't just keep adding Skin without making sure that someone is dedicated to maintain it long-term, because it increases the maintenance burden.

**Comment by Be-ing** [comment]
Yes it does not look 100% identical to Traktor, but imitating Traktor's aethetics is the intention and it's close enough that I think many people would notice the resemblance.

**Comment by uklotzde** [comment]
@Holzhaus Traktor Pro 3 vs. Traktor DJ 2 App

**Comment by intialonso** [comment]
I understand. Its ok, buy before closing this I want to ask if you have thinked about a centralized way to share user made skins, something like the KDE theme store? or just a wiki page? If Im not wrong, right know you have to dive into the forums to find skins.

**Comment by uklotzde** [comment]
This is the reference look: https://www.native-instruments.com/products/traktor/dj-software/traktor-dj-2/

**Comment by ronso0** [comment]
Nice how this creates half wavefroms by making use of `<Align>top/bottom</Align>` node in WVisual.
:+1: 
we should keep this in mind. maybe we can use it in other skins, like when the waveform splitter is dragged to have small waveforms, or when waveforms shrink due to other expanding elements.

**Comment by esbrandt** [comment]
@intialonso 
> I understand. Its ok, buy before closing this I want to ask if you have thinked about a centralized way to share user made skins, something like the KDE theme store? or just a wiki page? If Im not wrong, right know you have to dive into the forums to find skins.

There was an attempt for coding a sharing platform in a 2013 GSOC project.
Unfortunately it was left unfinished, and never gained traction. 

``The server-side component of Mixxx Backstage -- a new feature allowing users to share and customizations with other Mixxx users.``

https://github.com/mixxxdj/backstage

**Pull request closed without merging**